exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -89,6 +93,25 @@ type ManagedMachinePoolScaling struct {
 	MaxSize *int32 `json:"maxSize,omitempty"`
 }
 
+// TaintEffect is the effect for a Kubernetes taint.
+type TaintEffect string
+
+type Taint struct {
+	// Effect specifies the effect for the taint
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=NoSchedule;NoExecute;PreferNoSchedule
+	Effect TaintEffect `json:"effect"`
+	// Key is the key of the taint
+	// +kubebuilder:validation:Required
```

Comment:
do we need to explicitly mark it as Required? Kubebuilder should mark it as required as long as it doesn't have an `// +optional` marker

Reply:
will remove this one